contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -1276,6 +1276,17 @@ TsqlForXMLMakeFuncCall(TSQL_ForClause* forclause)
 						   root_name ? makeStringConst(root_name, -1) : makeStringConst("", -1));
 	fc = makeFuncCall(func_name, func_args, COERCE_EXPLICIT_CALL, -1);
 
+	/* In SQL Server if the result is empty then 0 rows are returned. Unfortunately it is not
+	 * possible to mimic this behavior solely using an aggregate, so we use an additional SRF
+	 * and pass the result to that function so that returning 0 rows is possible.
+	 */
+	func_name= list_make2(makeString("sys"), 
+							makeString(return_xml_type ? 
+								"tsql_select_for_xml_result" : 
+								"tsql_select_for_xml_text_result"));
```

Comment:
Can we have test cases for both `tsql_select_for_xml_result` and `tsql_select_for_xml_text_result` cases with 0 row output?

Reply:
sure, the current test is for text, let me add a test for xml as well